syntax = "proto3";
package Protocol;

// =============================================================
// Authentication
// =============================================================
message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  uint64 session_id = 2;
  float initial_x = 3;
  float initial_y = 4;
}

// =============================================================
// Game Logic
// =============================================================
message CS_Move {
  float x = 1;
  float y = 2;
  // TODO: Add timestamp or sequence for interpolation?
}

message SC_Move {
  uint64 entity_id = 1;
  float x = 2;
  float y = 3;
}

message SC_EnterRoom {
  int32 room_id = 1;
  // Repeated Entity info could go here
}
